src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java

```diff
@@ -752,8 +756,12 @@ private static int calculateTagColumn(Matcher javadocTagMatcher,
                 if (!isInIgnoreBlock(blockAst, throwAst)) {
                     final DetailAST newAst = throwAst.getFirstChild().getFirstChild();
                     if (newAst.getType() == TokenTypes.LITERAL_NEW) {
-                        final FullIdent ident = FullIdent.createFullIdent(newAst.getFirstChild());
-                        final ExceptionInfo exceptionInfo = new ExceptionInfo(
+                        DetailAST child = newAst.getFirstChild();
+                        final FullIdent ident = FullIdent.createFullIdent(child);
+                        while (child.getType() == TokenTypes.DOT) {
+                            child = child.getFirstChild();
+                        }
```

Comment:
Same question as above, what's the purpose of this `while`?

Reply:
same special case mentioned above~

Reply:
It looks like desire to keep the same column for violation.

```
$ cat Test.java 
class Test {
    public void bar() {
        throw new org.apache.tools.ant.BuildException(""); //warn
    }
}

$java -jar /var/tmp/checkstyle-8.32-all.jar -t Test.java 
CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|--LITERAL_CLASS -> class [1:0]
|--IDENT -> Test [1:6]
`--OBJBLOCK -> OBJBLOCK [1:11]
    |--LCURLY -> { [1:11]
    |--METHOD_DEF -> METHOD_DEF [2:4]
    |   |--MODIFIERS -> MODIFIERS [2:4]
    |   |   `--LITERAL_PUBLIC -> public [2:4]
    |   |--TYPE -> TYPE [2:11]
    |   |   `--LITERAL_VOID -> void [2:11]
    |   |--IDENT -> bar [2:16]
    |   |--LPAREN -> ( [2:19]
    |   |--PARAMETERS -> PARAMETERS [2:20]
    |   |--RPAREN -> ) [2:20]
    |   `--SLIST -> { [2:22]
    |       |--LITERAL_THROW -> throw [3:8]
    |       |   |--EXPR -> EXPR [3:14]
    |       |   |   `--LITERAL_NEW -> new [3:14]
    |       |   |       |--DOT -> . [3:38]
    |       |   |       |   |--DOT -> . [3:34]
    |       |   |       |   |   |--DOT -> . [3:28]
    |       |   |       |   |   |   |--DOT -> . [3:21]
    |       |   |       |   |   |   |   |--IDENT -> org [3:18]
    |       |   |       |   |   |   |   `--IDENT -> apache [3:22]
    |       |   |       |   |   |   `--IDENT -> tools [3:29]
    |       |   |       |   |   `--IDENT -> ant [3:35]
    |       |   |       |   `--IDENT -> BuildException [3:39]
    |       |   |       |--LPAREN -> ( [3:53]
    |       |   |       |--ELIST -> ELIST [3:54]
    |       |   |       |   `--EXPR -> EXPR [3:54]
    |       |   |       |       `--STRING_LITERAL -> "" [3:54]
    |       |   |       `--RPAREN -> ) [3:56]
    |       |   `--SEMI -> ; [3:57]
    |       `--RCURLY -> } [4:4]
    `--RCURLY -> } [5:0]
```
@rdiachenko ,  it is hard choice on where to put violation `org [3:18]` vs `BuildException [3:39]`

Reply:
I agree to keep column unchanged.
@HuGanghui , please make a static private method to get node where class name starts. Please generate diff regression report, as code change is significant.

Reply:
done, diff report will update later